diff --git a/runtime/http.js b/runtime/http.js
index bff10ecdad20c7b5cf7cdb3a468254584e4f0a13..d89d0868e5cb6e68816711508846db6dbfc08959 100644
--- a/runtime/http.js
+++ b/runtime/http.js
@@ -143,6 +143,11 @@ export function toWebRequest(/** @type {import('h3').H3Event} */ event) {
 		request: toWebRequestH3(event),
 		url: getRequestURL(event),
 	};
+
+	if (event.context.cf) {
+		event.web.request.cloudflare = event.context.cloudflare;
+	}
+
 	return event.web.request;
 }
 
